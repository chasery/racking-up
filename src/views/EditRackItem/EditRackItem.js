import React from "react";
import Form from "../../components/Form/Form";
import FormField from "../../components/FormField/FormField";
import "./EditRackItem.css";

function EditRackItem(props) {
  return (
    <main role="main">
      <section className="EditRackItem">
        <div className="EditRackItem__wrapper">
          <Form id="EditRackItem">
            <h2>Edit Rack Item</h2>
            <p>Edit the information for the clothing item.</p>
            <FormField id="name" label="Name" type="text" isRequired={true} />
            <FormField id="url" label="URL" type="text" isRequired={true} />
            <FormField id="price" label="Price" type="text" isRequired={true} />
            <button>Cancel</button>
            <button>Edit Rack Item</button>
          </Form>
        </div>
      </section>
    </main>
  );
}

export default EditRackItem;
