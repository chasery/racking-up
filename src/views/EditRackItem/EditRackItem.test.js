import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditRackItem from './EditRackItem';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    rackId: 1,
    rackItemId: 1,
  }),
}));

describe('EditRackItem component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<EditRackItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
