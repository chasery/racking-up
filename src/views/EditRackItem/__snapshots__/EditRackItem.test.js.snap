// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRackItem component renders without error 1`] = `
<main
  role="main"
>
  <section
    className="EditRackItem"
  >
    <div
      className="EditRackItem__wrapper"
    >
      <Form
        id="EditRackItem"
      >
        <h2>
          Edit Rack Item
        </h2>
        <p>
          Edit the information for the clothing item.
        </p>
        <FormField
          id="name"
          isRequired={true}
          label="Name"
          onChange={[Function]}
          type="text"
          value=""
        />
        <FormField
          id="url"
          isRequired={true}
          label="URL"
          onChange={[Function]}
          type="text"
          value=""
        />
        <FormField
          id="price"
          isRequired={true}
          label="Price"
          onChange={[Function]}
          type="text"
          value=""
        />
        <button>
          Cancel
        </button>
        <button>
          Edit Rack Item
        </button>
      </Form>
    </div>
  </section>
</main>
`;
